plugins/modules/purefb_proxy.py

```diff
@@ -67,55 +67,74 @@
 RETURN = r"""
 """
 
-HAS_PURITY_FB = True
+HAS_PYPURECLIENT = True
 try:
-    from purity_fb import Support
+    from pypureclient.flashblade import Support
 except ImportError:
-    HAS_PURITY_FB = False
+    HAS_PYPURECLIENT = False
 
 
 from ansible.module_utils.basic import AnsibleModule
 from ansible_collections.purestorage.flashblade.plugins.module_utils.purefb import (
-    get_blade,
+    get_system,
     purefb_argument_spec,
 )
 
 
 def delete_proxy(module, blade):
     """Delete proxy settings"""
-    changed = False
-    current_proxy = blade.support.list_support().items[0].proxy
-    if current_proxy != "":
-        changed = True
-        if not module.check_mode:
-            try:
-                proxy_settings = Support(proxy="")
-                blade.support.update_support(support=proxy_settings)
-            except Exception:
-                module.fail_json(msg="Delete proxy settigs failed")
+    changed = True
+    if not module.check_mode:
+        res = blade.support.update_support(support=Support(proxy=""))
+        if res.status_code != 200:
+            module.fail_json(
+                msg="Delete proxy settings failed. Error: {0}".format(
+                    res.errors[0].message
+                )
+            )
     module.exit_json(changed=changed)
 
 
 def create_proxy(module, blade):
     """Set proxy settings"""
-    changed = False
-    current_proxy = blade.support.list_support().items[0].proxy
+    changed = True
+    current_proxy = list(blade.get_support().items[0]).proxy
     if module.params["secure"]:
         protocol = "https://"
     else:
         protocol = "http://"
-    if current_proxy is not None:
-        changed = True
-        if not module.check_mode:
-            new_proxy = (
-                protocol + module.params["host"] + ":" + str(module.params["port"])
+    if not module.check_mode:
+        new_proxy = protocol + module.params["host"] + ":" + str(module.params["port"])
+        res = blade.patch_support(support=Support(proxy=new_proxy))
+        if res.status_code != 200:
+            module.fail_json(
+                msg="Set phone home proxy failed. Error: {0}".format(
+                    res.errors[0].message
+                )
             )
-            if new_proxy != current_proxy:
-                try:
-                    proxy_settings = Support(proxy=new_proxy)
-                    blade.support.update_support(support=proxy_settings)
-                except Exception:
-                    module.fail_json(msg="Set phone home proxy failed.")
+
+    module.exit_json(changed=changed)
+
+
+def update_proxy(module, blade):
+    """Update proxy settings"""
+    changed = False
+    current_proxy = list(blade.get_support().items[0]).proxy
```

Comment:
there is a mismatch in parenthesis closing, it should be like this **list(blade.get_support().items)[0].proxy**